pkg/ndctl/namespace.go

```diff
@@ -453,32 +453,27 @@ func (ns *Namespace) nullify() error {
 	var file *os.File
 
 	C.ndctl_namespace_set_raw_mode(ndns, 1)
+	defer C.ndctl_namespace_set_raw_mode(ndns, 0)
 	if err = ns.Enable(); err != nil {
-		C.ndctl_namespace_set_raw_mode(ndns, 0)
 		return err
 	}
 
-	if rc = C.posix_memalign(&buf, C.size_t(kib), C.size_t(kib)); rc < 0 {
-		C.ndctl_namespace_set_raw_mode(ndns, 0)
+	if rc = C.posix_memalign(&buf, C.size_t(kib4), C.size_t(kib4)); rc < 0 {
 		return fmt.Errorf("memory error: %s", cErrorString(rc))
 	}
+	defer C.free(buf)
+	C.memset(buf, 0, C.size_t(kib4))
 
-	file, err = os.OpenFile(ns.BlockDeviceName(), os.O_RDWR|syscall.O_DIRECT|os.O_EXCL, os.ModeDevice)
+	file, err = os.OpenFile("/dev/"+ns.BlockDeviceName(), os.O_RDWR|syscall.O_DIRECT|os.O_EXCL, os.ModeDevice)
 	if err != nil {
-		C.ndctl_namespace_set_raw_mode(ndns, 0)
 		return err
 	}
+	defer file.Close()
```

Comment:
Note, there is only one return point after defer file.Close, but yes we can have it for uniform style

Reply:
ya, you are write, we can avoid this defer.